code/modules/power/lighting.dm

```diff
@@ -625,10 +626,10 @@
 		else
 			L = new M.dispensing_bulb()
 		if(inserted_lamp)
-			if (current_lamp.light_status == LIGHT_OK && current_lamp.name == L.name) //name because I want this to be able to replace working lights with different colours
+			if (current_lamp.light_status == LIGHT_OK && current_lamp.name == L.name && brightness == initial(brightness) && current_lamp.color_r == L.color_r && current_lamp.color_g == L.color_g && current_lamp.color_b == L.color_b && on == has_power())
```

Comment:
what's with `on == has_power()`

Reply:
Ion Storm can turn off individual lights, and cyborgs can't pull the tubes out and put them back in like humans can